### PR TITLE
Added an option to timeout a request after x seconds

### PR DESCRIPTION
**Pull request by oakesja**
https://indigobio.atlassian.net/browse/AR-3141

### CODE REVIEW

**Comment by grimesjm** [comment]
cough cough https://github.com/yammer/circuitbox

File: lib/hyperion/hyperion.rb [hunk 1]
```diff
@@ -15,12 +15,13 @@ class Hyperion
...
   # @param additional_headers [Hash] headers to send in addition to the ones
   #   already determined by the route. Example: +{'User-Agent' => 'Mozilla/5.0'}+
+  # @param timeout [Integer] The limit of the entire request in seconds
   # @yield [result] yields the result if a block is provided
   # @yieldparam [HyperionResult]
   # @return [HyperionResult, Object] If a block is provided, returns the block's
   #   return value; otherwise, returns the result.
-  def self.request(route, body=nil, additional_headers={}, &block)
-    self.new(route).request(body, additional_headers, &block)
+  def self.request(route, body: nil, additional_headers: {}, timeout: 0, &block)
```
Comment (wintonpc) [inline lib/hyperion/hyperion.rb:23]: The change from optional params to keywords is going to break call sites. Have you identified how many and in which projects? Are you prepared to deal with the fallout from that?

**Comment by wintonpc** [comment]
Were you able to confirm that MzWeb aborts conversion when the client times out in this manner?

**Comment by oakesja** [comment]
Cody and I confirmed it with curl, which typhoeus wraps correct?

File: spec/lib/hyperion/requestor_spec.rb [hunk 2]
```diff
@@ -53,6 +53,12 @@ def assert_result(expected_result)
     expect(@request.body).to eql({'the' => 'body'})
   end
 
+  it 'makes requests with a timeout' do
```
Comment (wintonpc) [inline spec/lib/hyperion/requestor_spec.rb:64]: Where's the test for an actual time out?

**Review by wintonpc — CHANGES_REQUESTED** [review]

**Comment by wintonpc** [comment]
@oakesja (re: MzWeb) great! that'll work.

File: lib/hyperion/hyperion.rb (hunk 1 above)
Comment (oakesja) [reply lib/hyperion/hyperion.rb:23]: There are no breaking call sites that I could find. Web, Quartermaster, Nexus, Contra, and Assaymatic do not use `Hyperion.request` directly. They go through `Hyperion::Requestor` instead.

File: spec/lib/hyperion/requestor_spec.rb (hunk 2 above)
Comment (oakesja) [reply spec/lib/hyperion/requestor_spec.rb:64]: Added

File: lib/hyperion/hyperion.rb (hunk 1 above)
Comment (wintonpc) [reply lib/hyperion/hyperion.rb:23]: great

**Review by wintonpc — APPROVED** [review]

**Pull request merged**